cmd/simulacra/simulacra.go

```diff
@@ -178,43 +158,127 @@ func configureFromGCloud(project *string, zone *string) error {
 	flag.Parse()
 
-	if err := configureFromGCloud(project, zone); err != nil {
-		logger.Fatalf("project and zone must either be non empty or set in GCloud %v", err)
+	if *configPath != "" {
+		return getConfigFromYaml(*configPath)
+	}
+
+	config := Config{Platform: *platform, ConfigFilePath: *opsAgentConfigFile, Project: *project, Zone: *zone, Name: *name,
+		ThirdPartyAppsPath: *thirdPartyAppsPath}
+
+	return config, nil
+
+}
+
+func runCustomScripts(ctx context.Context, vm *gce.VM, logger *logging.DirectoryLogger, scripts []string) error {
+	for _, scriptPath := range scripts {
+		if scriptContent, err := os.ReadFile(scriptPath); err != nil {
+			return err
+		} else {
+			logger.ToMainLog().Printf("Running script from %s", scriptPath)
+			log.Default().Printf("Running script from %s", scriptPath)
+			if _, err := gce.RunScriptRemotely(ctx, logger, vm, string(scriptContent), nil, make(map[string]string)); err != nil {
+				return err
+			}
+			logger.ToMainLog().Printf("Done Running Script from  %s", scriptPath)
+
+		}
+	}
+	return nil
+}
+
+func main() {
+	loggingDir := fmt.Sprintf("/tmp/simulacra-%s", uuid.NewString())
+	logger, err := logging.NewDirectoryLogger(loggingDir)
+	if err != nil {
+		log.Default().Fatalf("Error initializing directory logger %v", err)
+	}
+	log.Default().Printf("Starting Simulacra, Detailed logging can be found in %s directory", loggingDir)
+	ctx := context.Background()
+	config, err := getSimulacraConfig()
+	if err != nil {
+		log.Default().Fatalf("error parsing simulacra config %v", err)
+	}
+
+	if err := configureFromGCloud(&config.Project, &config.Zone); err != nil {
+		log.Default().Fatalf("project and zone must either be non empty or set in GCloud %v", err)
 	}
 
 	options := gce.VMOptions{
-		Platform:    *platform,
-		MachineType: agents.RecommendedMachineType(*platform),
-		Name:        *name,
-		Project:     *project,
-		Zone:        *zone,
+		Platform:    config.Platform,
+		MachineType: agents.RecommendedMachineType(config.Platform),
+		Name:        config.Name,
+		Project:     config.Project,
+		Zone:        config.Zone,
 	}
 	// Create VM Instance.
-	vm, err := gce.CreateInstance(ctx, logger, options)
+	log.Default().Printf("Creating VM Instance, check %s for details", vmInitLogFile)
+	vm, err := gce.CreateInstance(ctx, logger.ToFile(vmInitLogFile), options)
 	if err != nil {
-		logger.Fatalf("Failed to create GCE instance %v", err)
+		log.Default().Fatalf("Failed to create GCE instance %v", err)
 	}
 	// Install Ops Agent on VM.
-	if err := setupOpsAgent(ctx, vm, logger, *configFile); err != nil {
-		logger.Fatalf("Failed to install Ops Agent %v", err)
+	log.Default().Print("Installing Ops Agent, check main_log.txt for details")
```

Comment:
Add the full log file path here, if I use simulacra several times, the log files for each are going to separate directories. Just add `loggingDir` to the log here